### DISCUSSION TITLE
Is it possible to render thousands of annotation in real time?

### DISCUSSION THREAD

**Discussion by Songhao-LI** [annotorious/annotorious#473] [Q&A]
Hi all, when I use openseadragon and annotorious 3.x to render thousands of annotations, I found it may cause a terrible UI lagging. I am not sure how to fix it. Has anyone met this issue?🙏🙏🙏

**Comment by rsimon** [comment]
This might depend on many aspects. Your exact implementation, complexity of annotations, box vs. polygon, etc. But I've been successfully using it to render up to 80k annotations. Can you share an online example?
